src/input/input_json.rs

```diff
@@ -89,10 +89,22 @@ impl<'a> Input<'a> for JsonInput {
             _ => Err(ValError::new(ErrorTypeDefaults::StringType, self)),
         }
     }
-    fn lax_str(&'a self) -> ValResult<EitherString<'a>> {
+    fn lax_str(&'a self, coerce_numbers_to_str: bool) -> ValResult<EitherString<'a>> {
         match self {
             JsonInput::String(s) => Ok(s.as_str().into()),
-            _ => Err(ValError::new(ErrorTypeDefaults::StringType, self)),
+            _ => {
```

Comment:
You could also write

```suggestion
            JsonInput::BigInt(v) if coerce_numbers_to_str => Ok(v.to_string().into())
            // and same for other number types ...
            _ => {
```

You'd need to add `impl From<String> for EitherString<'_>`, I think.